notation.go

```diff
@@ -42,6 +42,17 @@ type SignOptions struct {
 
 	// SigningAgent sets the signing agent name
 	SigningAgent string
+
+	// OCIimageManifest is the flag to specify if user wants to use OCI image
+	// manifest to store signatures in remote registries.
+	// By default, Notation will use OCI artifact manifest to store signatures.
+	// If OCIimageManifest flag is set to true, Notation will instead use
+	// OCI image manifest.
+	// Note, Notation will not automatically convert between these two types
+	// on any occasion.
+	// OCI artifact manifest: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/artifact.md
+	// OCI image manifest: https://github.com/opencontainers/image-spec/blob/v1.1.0-rc2/manifest.md
+	OCIimageManifest bool
```

Comment:
To push the signature using OCI image manifest or OCI artifact manifest is not related to the `signer.Sign()` itself. We should introduce `RemoteSignOptions` like `RemoteVerifyOptions`. However, this solution is still not appropriate since whether a repository supports OCI artifact manifest is a property of the `registry.Repository`.

Therefore, I would like to propose the following changes in the package `registry`:
```go
// repositoryClient implements Repository
type repositoryClient struct {
	registry.Repository
	RepositoryOptions 
}

type RepositoryOptions struct {
	PushImageManifest bool
}

func NewRepositoryWithOptions(repo registry.Repository, opts RepositoryOptions) Repository {
	return &repositoryClient{
		Repository:        repo,
		RepositoryOptions: opts,
	}
}

```